<?xml version="1.0"?>
<launch>
  
  <arg name="info_topic"      default="/camera/color/camera_info" />
  <arg name="img_base_topic"  default="/camera/color/image_raw" />
  <arg name="img_transport"   default="raw" />

  <arg name="circle_diameter" default="0.212" />
  <arg name="id_bits"         default="3" />
  <arg name="id_samples"      default="720" />
  <arg name="hamming_dist"    default="1" />
  <arg name="num_markers"     default="1" />
  <arg name="use_gui"         default="true" />
  <arg name="min_size"        default="50" />
  
  <arg name="calib_file"      default="" />
  <arg name="coords_method"   default="0" />



  <node pkg="whycode" exec="whycode_node" name="whycode_node">
    <param name="info_topic"      value="$(var info_topic)" />
    <param name="img_base_topic"  value="$(var img_base_topic)" />
    <param name="img_transport"   value="$(var img_transport)" />
    
    <param name="circle_diameter" value="$(var circle_diameter)" />
    <param name="id_bits"         value="$(var id_bits)" />
    <param name="id_samples"      value="$(var id_samples)" />
    <param name="hamming_dist"    value="$(var hamming_dist)" />
    <param name="num_markers"     value="$(var num_markers)" />
    <param name="use_gui"         value="$(var use_gui)" />
    <param name="min_size"        value="$(var min_size)" />
    
    <param name="calib_file"      value="$(var calib_file)" />
    <param name="coords_method"   value="$(var coords_method)" />
  </node>

</launch>
